Source/ACE/Entity/Player.cs

```diff
@@ -194,6 +194,43 @@ public void HandleActionLearnSpell(uint spellId)
             learnSpellChain.EnqueueChain();
         }
 
+        /// <summary>
+        /// This method implements player spell bar management for - adding a spell to a specific spell bar (0 based) at a specific slot (0 based).
+        /// </summary>
+        /// <param name="spellId"></param>
+        /// <param name="spellBarPositionId"></param>
+        /// <param name="spellBarId"></param>
+        public void HandleActionAddSpellToSpellBar(uint spellId, uint spellBarPositionId, uint spellBarId)
+        {
+            // The spell bar magic happens here. First, let's mind our race conditions....
+            // No need for action chain here - internal to us, no other object can change us.
+            AceObject.SpellsInSpellBars.Add(new AceObjectPropertiesSpellBarPositions()
+                                                { AceObjectId = AceObject.AceObjectId,
+                                                  SpellId = spellId,
+                                                  SpellBarId = spellBarId,
+                                                  SpellBarPositionId = spellBarPositionId });
+        }
+
+        /// <summary>
+        /// This method implements player spell bar management for - removing a spell to a specific spell bar (0 based)
+        /// </summary>
+        /// <param name="spellId"></param>
+        /// <param name="spellBarId"></param>
+        public void HandleActionRemoveSpellToSpellBar(uint spellId, uint spellBarId)
+        {
+            // More spell bar magic happens here. First, let's mind our race conditions....
+            // No need for action chain here - internal to us, no other object can change us.
+            AceObject.SpellsInSpellBars.Remove(AceObject.SpellsInSpellBars.Single(x => x.SpellBarId == spellBarId && x.SpellId == spellId));
+            // Now I have to reorder
+            var sorted = AceObject.SpellsInSpellBars.FindAll(x => x.AceObjectId == AceObject.AceObjectId && x.SpellBarId == spellBarId).OrderBy(s => s.SpellBarPositionId);
```

Comment:
I'm pretty strongly opinionated that there should be a logical separation between the two components (database/backend, and in-game data).  I would really like to see this abstracted consistently with the remainder of the data that falls into this paradigm.  If your strongly against doing so I wont hold up this PR over it, but the inner me that loves consistency and well defined interfaces will be very sad.